bark/tools/psg_view.py

```diff
@@ -0,0 +1,898 @@
+from PyQt5.QtGui import QIcon
+from PyQt5 import QtCore, QtWidgets
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QWidget, QInputDialog, QLineEdit, QFileDialog
+from PyQt5.QtWidgets import QApplication, QMainWindow, QSizePolicy, QWidget, QComboBox, QLabel, QRadioButton, QCheckBox, QGridLayout, QLineEdit, QScrollArea
+
+ 
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
+from matplotlib.figure import Figure 
+import matplotlib.pyplot as plt
+
+import os
+import sys
+import string
+import yaml
+import numpy as np
+
+import bark
+from bark.io.eventops import (OpStack, write_stack, read_stack, Update, Merge,
+                              Split, Delete, New)
+import warnings
+warnings.filterwarnings('ignore')  # suppress matplotlib warnings
+
+
+help_string = '''
+Pressing any number or letter (uppercase or lowercase) will mark a segment.
+
+Shortcuts
+---------
+any letter or number    annotate segment
+ctrl+s                  saves the annotation data
+ctrl+h                  prints this message
+ctrl+o                  zoom out
+ctrl+i                  zoom in
+ctrl+j,right            next segment
+ctrl+i,left             previous segment
+
+ctrl+z                  undo last operation
+ctrl+y                  redo
+ctrl+w                  close
+
+click on the right of the current label        next segment
+click on the left of the current label        previous segment
+
+The bottom panel is a map of all label locations.
+Click on a label to travel to that location.
+
+On close, an operation file and the final event file will be written.
+Do not kill from terminal unless you want to prevent a save.
+
+To create custom label, input the label name in the right toolbar.
+
+
+''' 
+
+
+color = 'green'
+fontsize = 12
+# kill all the shorcuts
+def kill_shortcuts(plt):
+    plt.rcParams['keymap.all_axes'] = ''
+    plt.rcParams['keymap.back'] = ''
+    plt.rcParams['keymap.forward'] = ''
+    plt.rcParams['keymap.fullscreen'] = ''
+    plt.rcParams['keymap.grid'] = ''
+    plt.rcParams['keymap.home'] = ''
+    plt.rcParams['keymap.pan'] = ''
+    #plt.rcParams['keymap.quit'] = ''
+    plt.rcParams['keymap.save'] = ''
+    plt.rcParams['keymap.xscale'] = ''
+    plt.rcParams['keymap.yscale'] = ''
+    plt.rcParams['keymap.zoom'] = ''
+
+
+def labels_to_scatter_coords(labels):
+    times = [x['start'] for x in labels]
+    values = []
+    for record in labels:
+        name = record['name']
+        if not isinstance(name, str) or name == '':
+            v = 0
+        elif name.isdigit():
+            v = int(name)
+        elif name[0].isalpha():
+            # alphabet in range 11-36
+            v = 133 - ord(name[0].lower())
+        else:
+            v = 37
+        values.append(v)
+    return times, values
+
+
+def nearest_label(labels, xdata):
+    return np.argmin(np.abs(xdata - np.array([x['start'] for x in labels])))
+
+
+def write_metadata(path, meta='.meta.yaml'):
```

Comment:
This function should call `bark.write_metadata` rather than reproduce its functionality.